Library/Homebrew/global.rb

```diff
@@ -26,7 +26,7 @@
 end
 RUBY_BIN = RUBY_PATH.dirname
 
-HOMEBREW_USER_AGENT = "Homebrew #{HOMEBREW_VERSION} (Ruby #{RUBY_VERSION}-#{RUBY_PATCHLEVEL}; #{OS_VERSION})"
+HOMEBREW_USER_AGENT = "#{ENV["HOMEBREW_USER_AGENT"]} (Ruby #{RUBY_VERSION}-#{RUBY_PATCHLEVEL}; #{OS_VERSION})"
```

Comment:
One big Ruby 1.x vs 2.x change is how it handles character encodings, which for HTTP seems more germane than OS version, processor arch, or even `curl` version. (Or is that Ruby 1.8 vs 1.9?)

I've seen core PRs that get "we can't do this, b/c of Ruby 1.8 compatibility" line notes. They mostly seem like "it'll throw an error" b/c of syntax differences or unimplemented features before it gets to the network request phase. But maybe there are externally-vislble behavior differences, too. (`Brew` mostly defers to `curl` and `git` for network operations, but maybe stuff will make it in to its command line arguments, if not vetted strictly.)

Maybe conditionally append a `; Ruby 1.x.y` for the old-Ruby cases?

"Cross your fingers and just wait for it to go away" seems reasonable to me too, and that would simplify the header comment format.